src/refract/InfoElements.h

```diff
@@ -0,0 +1,86 @@
+//
+//  refract/InfoElements.h
+//  librefract
+//
+//  Created by Thomas Jandecka on 21/08/2017
+//  Copyright (c) 2017 Apiary Inc. All rights reserved.
+//
+
+#ifndef REFRACT_INFO_ELEMENTS_H
+#define REFRACT_INFO_ELEMENTS_H
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "ElementIfc.h"
+
+namespace refract
+{
+    class InfoElements final
+    {
+        using Container = std::vector<std::pair<std::string, std::unique_ptr<IElement> > >;
+        Container elements;
+
+    public:
+        using iterator = typename Container::iterator;
+        using const_iterator = typename Container::const_iterator;
+
+    public:
+        InfoElements() = default;
+        ~InfoElements() = default;
+
+        InfoElements(const InfoElements& other);
+
+        InfoElements(InfoElements&&) = default;
+
+        InfoElements& operator=(const InfoElements& rhs);
+        InfoElements& operator=(InfoElements&&) = default;
+
+    public:
+        const_iterator begin() const noexcept
+        {
+            return elements.begin();
+        }
+
+        // FIXME potential untracked modification
+        iterator begin() noexcept
+        {
+            return elements.begin();
+        }
+
+        const_iterator end() const noexcept
+        {
+            return elements.end();
+        }
+
+        // FIXME potential untracked modification
+        iterator end() noexcept
+        {
+            return elements.end();
+        }
+
+        const_iterator find(const std::string& name) const;
+        iterator find(const std::string& name);
+
+        IElement& set(const std::string& key, std::unique_ptr<IElement> value);
+        IElement& set(const std::string& key, const IElement& value);
+
+        /// clone elements from `other` to `this`
+        void clone(const InfoElements& other);
```

Comment:
why are `beging/end` methods imlementation in header, while `erase/clear/empty/size` are in `cc` file?
All of them just forwarding call to member `elements`.

Reply:
I agree there is inconsistency. What about the other option, putting everything out of the header? Introducing interfaces.

Reply:
Have not problem with move to cc. Choose one of them by your feeling :)